app/utils/records-list.js

```diff
@@ -50,6 +50,12 @@ export default {
         let targetModelName = get(target, 'constructor.modelName')
                            || get(target, 'content.constructor.modelName');
 
+        // If the relationship isn't there yet
+        if (!relationshipId && relationshipModelName && targetId && targetModelName) {
+          let foundId = found.belongsTo(targetModelName.camelize()).id();
+          return foundId == targetId;
+        }
```

Comment:
Not sure if I'm a fan of having to do this. Definitely needs a test. But only way I could get this to work for hasMany relationships where one model `isNew`.